contracts/governance/Vesting/fouryear/FourYearVesting.sol

```diff
@@ -55,10 +56,26 @@ contract FourYearVesting is FourYearVestingStorage, Proxy {
 	}
 
 	/**
-	 * @dev We need to add this implementation to prevent proxy call FourYearVestingLogic.governanceWithdrawTokens
+	 * @dev We need to add this implementation to prevent proxy call
+	 * FourYearVestingLogic.governanceWithdrawTokens.
 	 * @param receiver The receiver of the token withdrawal.
 	 * */
-	function governanceWithdrawTokens(address receiver) public {
+	function governanceWithdrawTokens(address receiver) public pure {
 		revert("operation not supported");
 	}
+
+	/**
+	 * @notice Set address of the implementation - vesting owner.
+	 * @dev Overriding setImplementation function of UpgradableProxy. The logic can only be
+	 * modified when both token owner and veting owner approve. Since
+	 * setImplementation can only be called by vesting owner, we also need to check
+	 * if the new logic is already approved by the token owner.
+	 * @param _implementation Address of the implementation. Must match with what is set by token owner.
+	 * */
+	function setImplementation(address _implementation) public onlyProxyOwner {
+		require(newImplementation != address(0), "invalid address");
```

Comment:
pls use oz Address library to check if the address is a contract or use EIP-165 to verify if the contract supports expected interfaces

Reply:
Done